wordhelper/anagram.py

```diff
@@ -56,22 +86,64 @@ def find(self):
 				anagrams += words_by_character_frequency[histogram]
 		return anagrams
 
-	def find_multi(self, words_so_far=[]):
+	def find_multi_helper(self, use_words_until, words_so_far):
+		"""
+		Helper function for find_multi
+		:param use_words_until: will only search words between indices [0, use_words_until] inclusive in the dictionary. This prevents re-searching the same word
+		:type use_words_until: int
+		:param words_so_far: matched words so far. Used for printing solutions as we find them
+		:type words_so_far: list[string]
+		:rtype: list[list[string]]
+		"""
 		single_word_anagrams = self.find()
-		if len(single_word_anagrams) > 0:
-			for i in range(len(single_word_anagrams)):
-				single_word_anagrams[i] = [single_word_anagrams[i]]
-				print(words_so_far + single_word_anagrams[i])
-			return single_word_anagrams
+		largest_word = get_current_dictionary().words_sorted_by_length_then_lex[use_words_until]
+		single_word_anagrams_deduped = []
+		for single_word_anagrams in single_word_anagrams:
```

Comment:
I'm not sure how python is allowing this, but you've defined this thing twice. One should be singular